src/main/kotlin/net/axay/kspigot/chat/LiteralTextBuilder.kt

```diff
@@ -18,10 +23,10 @@ import net.md_5.bungee.api.chat.hover.content.Text
 inline fun literalText(
     baseText: String = "",
     builder: LiteralTextBuilder.() -> Unit = { }
-) = LiteralTextBuilder(baseText).apply(builder).build() as TextComponent
+) = LiteralTextBuilder(baseText).apply(builder).build()
 
-class LiteralTextBuilder(val internalText: BaseComponent, ) {
-    constructor(text: String) : this(TextComponent(text))
+class LiteralTextBuilder(val internalText: Component) {
```

Comment:
comment for this class in general: are you sure that the behaviour will stay the same here? the bungeecord text api had some weird special behaviour, which may not be the same with adventure components

Reply:
I tested it using all methods, style variables and even nested literalTexts so it should work as inteded

Reply:
alright, nested literal texts are the part which I was most interested in here